From dd36f7430eed1724cfa8bc30ceaa6159569ee9e8 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=D0=A0=D1=83=D1=81=D0=BB=D0=B0=D0=BD=20=D0=98=D0=B6=D0=B1?=
 =?UTF-8?q?=D1=83=D0=BB=D0=B0=D1=82=D0=BE=D0=B2?= <lrn1986@gmail.com>
Date: Tue, 14 Apr 2015 10:51:02 +0000
Subject: [PATCH 15/20] [W32] Try to use pkg-config to detect ICU libraries

MinGW packages tend to have .pc files these days. Use them.

ICU naming and structure is complex enough to warrant this, and asking
pkg-config might result in things like
UNICODE_LIBS="-licui18n54 -licuuc54 -licudata54"
instead of the default
UNICODE_LIBS="-licui18n -licuuc"
and pkg-config usually knows best.
---
 Source/autotools/FindDependencies.m4 | 10 ++++++++--
 1 file changed, 8 insertions(+), 2 deletions(-)

diff --git a/Source/autotools/FindDependencies.m4 b/Source/autotools/FindDependencies.m4
index 3ce3f76..a21e68d 100644
--- a/Source/autotools/FindDependencies.m4
+++ b/Source/autotools/FindDependencies.m4
@@ -109,8 +109,14 @@ case "$host" in
         UNICODE_LIBS="-licucore"
         ;;
     *-*-mingw*)
-        UNICODE_CFLAGS=""
-        UNICODE_LIBS="-licui18n -licuuc"
+	PKG_CHECK_MODULES(ICU, icu-i18n, ,)
+	if test "x$ICU_LIBS" = "x" ; then
+        	UNICODE_CFLAGS=""
+        	UNICODE_LIBS="-licui18n -licuuc"
+	else
+        	UNICODE_CFLAGS="$ICU_CFLAGS"
+        	UNICODE_LIBS="$ICU_LIBS"
+	fi
         AC_CHECK_HEADERS([unicode/uchar.h], [], [AC_MSG_ERROR([Could not find ICU headers.])])
         ;;
     *)
-- 
1.8.5.3

